### PR TITLE
Fix incorrect stylesheet order for all non-EN locales

### PR DESCRIPTION
**Pull request by marekhrabe**
This PR makes sure that dynamically loaded CSS is always loaded in intended order.

We had a bug in our codebase that could cause wrong order of stylesheets. That lead to overriding specific section styles (for example post-editor.css) by the general style.css. It looked like this for example:
 
screen shot 2017-11-01 at 14 14 34

The bug was race-conditioned and most importantly, was only reproducible in development in all locales other than English and also in production for RTL languages. 

The biggest problem was that all dynamically loaded stylesheets were appended to the bottom of `<head>`, regardless of needed order. This PR changes it in a way, that new CSS is inserted right after the original `<link rel="stylesheet">` that the new CSS is replacing. This way, we always have general application-wid CSS first and specific section CSS after it.

### CODE REVIEW

File: client/lib/i18n-utils/switch-locale.js [hunk 1]
```diff
@@ -79,9 +80,10 @@ export function switchCSS( elementId, cssUrl, callback = noop ) {
 /**
  * Loads a css stylesheet into the page
  * @param {string} cssUrl - a url to a css resource to be inserted into the page
+ * @param {Element} currentLink - a <link> DOM element that we want to use as a reference for stylesheet order
  * @param {Function} callback - a callback function to be called when the CSS has been loaded (after 500ms have passed).
  */
-function loadCSS( cssUrl, callback = noop ) {
+function loadCSS( cssUrl, currentLink = null, callback = noop ) {
```
Comment (stephanethomas) [inline client/lib/i18n-utils/switch-locale.js]: Why do we need to define a default value for `currentLink`?

File: server/pages/index.jade [hunk 2]
```diff
@@ -112,7 +115,7 @@ html(lang=lang, dir=isRTL ? 'rtl' : 'ltr', class=isFluidWidth ? 'is-fluid-width'
 			else
 				if sectionCss
 					link(rel='preload' as='style' href=sectionCss)
-		
+
 		if shouldUseStylePreloadCommon
 			if isRTL
 				link(rel='preload' as='style' href=urls['style-rtl.css'])
```
Comment (stephanethomas) [inline server/pages/index.jade]: Shouldn't we also update that part of the code as well?

We have other [`.jade templates`](https://github.com/Automattic/wp-calypso/tree/master/server/pages) that could benefit from this change as far I can see.

Comment (marekhrabe) [reply server/pages/index.jade]: I wasn't aware of other pages, thanks! I'll search over the whole codebase now for stylesheet links

File: client/lib/i18n-utils/switch-locale.js (hunk 1 above)
Comment (marekhrabe) [reply client/lib/i18n-utils/switch-locale.js]: no need, you are right

**Comment by marekhrabe** [comment]
I think the feedback was addressed. Ready for another testing/review round

**Comment by marekhrabe** [comment]
Regarding the testing instructions, that one is not 100% clear to me. There seemed to be some randomness involved.

Essentially:
- switch language to a non-default one (RTL preferred, but it should work with any)
- load /post/yoursite.wordpress.com
- you should see broken editor because `style.css` was appended to the `<head>` and overrides section-specific style

This, however, does not guarantee you broken editor all the time. It seemed more of an issue on calypso.localhost, calypso.live and in the desktop app but @bisko, I think, reproduced it also in production.

**Comment by bisko** [comment]
The best way to reproduce it is to open a post for editing normally and when it’s loaded if you don’t see a broken header, reload the page. This usually triggers it.

**Comment by stephanethomas** [comment]
The changes look good to me, and work fine. Nice work!

Note I was able to reproduce the issue with `calypso.live` and in development, but not in production. I wonder if this has anything to do with user bootstrapping being disabled for the former environments.

**Comment by marekhrabe** [comment]
We definitely had seen reports from Desktop app where users were experiencing this issue and reporting to HEs. I'll make sure to test this in next release so we are sure we are fixing the bug for everybody. Thanks for your review! :)

**Pull request merged**